src/approx_itensornetwork/approx_itensornetwork.jl

```diff
@@ -22,6 +22,8 @@ function approx_itensornetwork(
   partition_wo_deltas = _contract_deltas_ignore_leaf_partitions(
     binary_tree_partition; root=root
   )
+  density_matrix_alg =
+    alg isa Algorithm"density_matrix_direct_eigen" ? "direct_eigen" : "qr_svd"
```

Comment:
It seems like the main use for `Algorithm"density_matrix_direct_eigen"` is this selection of the `density_matrix_alg`. Maybe simpler to just make `density_matrix_alg` a keyword argument directly and remove `Algorithm"density_matrix_direct_eigen"`.